android/src/main/java/com/applovin/reactnative/AppLovinMAXAdViewUiComponent.java

```diff
@@ -144,7 +145,9 @@ public void onAdLoaded(@NonNull final MaxAd ad)
     {
         WritableMap adInfo = AppLovinMAXModule.getInstance().getAdInfo( ad );
 
-        AppLovinMAXModule.getInstance().sendReactNativeEvent( "OnNativeUIComponentAdviewAdLoadedEvent", adInfo );
+        // Copy adInfo since it is consumed once it is sent
```

Comment:
What do you mean "consumed"?

Reply:
The same object can't be set when sent to the JavaScript side. It looks like the object is queued internally in the same function.

Reply:
When the same object is sent consecutively via `receiveEvent()`, the following exception is thrown in the 2nd `receiveEvent()`.

```
com.facebook.react.bridge.ObjectAlreadyConsumedException: Map already consumed
	at com.facebook.react.bridge.WritableNativeArray.pushNativeMap(Native Method)
	at com.facebook.react.bridge.WritableNativeArray.pushMap(WritableNativeArray.java:57)
	at com.facebook.react.bridge.Arguments.fromJavaArgs(Arguments.java:187)
	at com.facebook.react.bridge.JavaScriptModuleRegistry$JavaScriptModuleInvocationHandler.invoke(JavaScriptModuleRegistry.java:83)
	at java.lang.reflect.Proxy.invoke(Proxy.java:1006)
	at $Proxy7.receiveEvent(Unknown Source)
	at com.applovin.reactnative.AppLovinMAXAdViewUiComponent.sendReactNativeCallbackEvent(AppLovinMAXAdViewUiComponent.java:242)
	at com.applovin.reactnative.AppLovinMAXAdViewUiComponent.onAdLoaded(AppLovinMAXAdViewUiComponent.java:155)
	at com.applovin.impl.bc.f(SourceFile:1735)
	at com.applovin.impl.bc.$r8$lambda$4NuWt1vJot1WeTfwJ9stHJza1pQ(Unknown Source:0)
	at com.applovin.impl.bc$$ExternalSyntheticLambda37.run(D8$$SyntheticClass:0)
	at android.os.Handler.handleCallback(Handler.java:958)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:205)
	at android.os.Looper.loop(Looper.java:294)
	at android.app.ActivityThread.main(ActivityThread.java:8177)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
```

So, the comment would be like this?

```
Copy adInfo - the same object can not be applied to receiveEvent() consecutively, otherwise ObjectAlreadyConsumedException is thrown.
```

Reply:
Yeah, I don't think we need the null check though, would be cleaner

```
        // Copy the `adInfo` since sending the same map through the RN bridge more than once will result in `com.facebook.react.bridge.ObjectAlreadyConsumedException: Map already consumed`
        AppLovinMAXModule.getInstance().sendReactNativeEvent( "OnNativeUIComponentAdViewAdLoadedEvent", adInfo.copy() );

        if ( containerView != null )
```

Or we can just make a copy of the `adInfo` passed into `sendReactNativeEvent(...)` so this issue never happens again in the future (not sure how heavy the `copy()` operation is though. Up to you!